src/main/java/org/kiwiproject/dropwizard/poller/health/ClientPollerHealthChecks.java

```diff
@@ -0,0 +1,96 @@
+package org.kiwiproject.dropwizard.poller.health;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static org.kiwiproject.base.KiwiStrings.format;
+import static org.kiwiproject.metrics.health.HealthCheckResults.newUnhealthyResultBuilder;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.google.common.annotations.VisibleForTesting;
+import lombok.experimental.UtilityClass;
+import org.kiwiproject.dropwizard.poller.metrics.ClientPollerStatistics;
+import org.kiwiproject.metrics.health.HealthStatus;
+
+import java.net.URI;
+import java.time.temporal.TemporalUnit;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+@UtilityClass
+public class ClientPollerHealthChecks {
+
+    private static final String UNSPECIFIED_URI_PREFIX = "[unspecified-URI-";
+    private static final String UNSPECIFIED_URI_SUFFIX = "]";
+    private static final Supplier<String> NAME_FOR_UNSPECIFIED_URI = ClientPollerHealthChecks::nameForUnspecifiedUri;
+
+    @VisibleForTesting
+    static final String FAILURE_DETAILS_KEY = "FailureDetails";
+
+    private static String nameForUnspecifiedUri() {
+        return UNSPECIFIED_URI_PREFIX + System.currentTimeMillis() + UNSPECIFIED_URI_SUFFIX;
+    }
+
+    static String humanReadableOf(TemporalUnit unit) {
```

Comment:
Can this be private? Or does it need `@ VisibleForTesting`?

Reply:
it is called from the health checks in the same package, that is why it is package private